src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -89,47 +114,51 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
     <hr/>
 
     <% if (request.getSession().getAttribute("user") != null) { %>
-    <form action="/chat/<%= conversation.getTitle() %>" method="POST" enctype="multipart/form-data">
+    <form name="chat" action="/chat/<%= conversation.getTitle() %>" method="POST" enctype="multipart/form-data">
         <input type="text" name="message">
+        <input type="hidden" name="GIFSrc" id="hiddenField" value=""/>
         <input type="file" name="upload">
+        <button type="submit">Send</button>
+    </form>
+
         <!-- Button trigger modal -->
-		    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#exampleModalLong">
+		    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#modal">
 		    Send a GIF
 		    </button>
-
+    	<form id="gifSearchForm" name="gifSearch" action="/search/" method="GET"> 
     		<!-- Modal -->
-    		<div class="modal fade" id="exampleModalLong" tabindex="-1" role="dialog" aria-labelledby="exampleModalLongTitle" aria-hidden="true">
+    		<div class="modal fade" id="modal" tabindex="-1" role="dialog" aria-labelledby="modalTitle" aria-hidden="true">
     		  <div class="modal-dialog" role="document">
     		    <div class="modal-content">
     		      <div class="modal-header">
-    		        <input type="search" class="form-control" id="GIFsearch" placeholder="Search GIFs on GIPHY...">
+    		        <input type="search" name="search" class="form-control" id="GIFsearch" placeholder="Search GIFs on GIPHY...">
     		        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
     		          <span aria-hidden="true">&times;</span>
     		        </button>
     		      </div>
     		      <div class="modal-body">
-    		        <div class="container">
+    		        <div id="modal-container" class="container">
     				  <div class="row">
     				    <div class="col">
     				    	<div data-dismiss="modal">
-         						<img src="https://i.giphy.com/media/feqkVgjJpYtjy/200.gif" class="img-fluid" alt="Responsive image">
+         						<img onclick="sendGIF(this)" src="https://i.giphy.com/media/feqkVgjJpYtjy/200.gif" class="img-fluid" alt="Responsive image">
```

Comment:
Ah, so these hardcoded gifs show up before the user searches, but then they get replaced with the search results?

Reply:
Exactly!